src/olympia/api/fields.py

```diff
@@ -33,56 +33,48 @@ class TranslationSerializerField(fields.WritableField):
 
     def __init__(self, *args, **kwargs):
         self.min_length = kwargs.pop('min_length', None)
-
         super(TranslationSerializerField, self).__init__(*args, **kwargs)
-        # Default to return all translations for each field.
         self.requested_language = None
 
-    def initialize(self, parent, field_name):
-        super(TranslationSerializerField, self).initialize(parent, field_name)
-        request = self.context.get('request', None)
-        if request and request.method == 'GET' and 'lang' in request.GET:
-            # A specific language was requested, we will only return one
-            # translation per field.
-            self.requested_language = request.GET['lang']
-
     def fetch_all_translations(self, obj, source, field):
         translations = field.__class__.objects.filter(
             id=field.id, localized_string__isnull=False)
         return dict((to_language(trans.locale), unicode(trans))
                     for trans in translations) if translations else None
 
-    def fetch_single_translation(self, obj, source, field):
+    def fetch_single_translation(self, obj, source, field, requested_language):
         return unicode(field) if field else None
 
-    def field_to_native(self, obj, field_name):
-        source = self.source or field_name
-        value = obj
-        for component in source.split('.'):
-            value = fields.get_component(value, component)
-            if value is None:
-                break
-
-        field = value
-        if field is None:
+    def get_attribute(self, obj, requested_language=None):
```

Comment:
I know this is lifted from zamboni's, but I don't see any calls to `get_attribute()` with `requested_language` set. Do we really need it here ? If not we could also remove the `if requested_language is None:` below (since it's always true).